frontend/src/app/game/game.service.ts

```diff
@@ -99,15 +99,15 @@ export class GameService {
       return;
     }
 
-    if (this.game.isFinished()) {
-      this.setSpeed('pause');
-      return;
-    }
-
     const gameUpdate = this.game.moveForward();
     const event = gameUpdate.event;
     this.showEvent(event);
 
+    if (this.game.isFinished()) {
+      this.setSpeed('finished');
```

Comment:
I do not see you would be saving the data. Could you, or where should that be made?